src/devices/video/ef9345.cpp

```diff
@@ -105,10 +107,25 @@ ef9345_device::ef9345_device(const machine_config &mconfig, const char *tag, dev
 	device_video_interface(mconfig, *this),
 	m_space_config("videoram", ENDIANNESS_LITTLE, 8, 16, 0, nullptr, *ADDRESS_MAP_NAME(ef9345)),
 	m_charset(*this, DEVICE_SELF),
+	m_variant(TYPE_EF9345),
 	m_palette(*this, finder_base::DUMMY_TAG)
 {
 }
 
+ef9345_device::ef9345_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock, uint32_t variant) :
+	device_t(mconfig, EF9345, "EF9345", tag, owner, clock, "ef9345", __FILE__),
+	device_memory_interface(mconfig, *this),
+	device_video_interface(mconfig, *this),
+	m_space_config("videoram", ENDIANNESS_LITTLE, 8, 16, 0, nullptr, *ADDRESS_MAP_NAME(ef9345)),
+	m_charset(*this, DEVICE_SELF),
+	m_variant(variant),
+	m_palette(*this, finder_base::DUMMY_TAG)
+{
+}
+
+ts9347_device::ts9347_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
```

Comment:
This won't work properly - the device will still have the EF9345 type/name/shortname.  It should even fail the `mame -valid` checks.  You need to make the base class constructor pass this info through as well.